common/src/api/external/mod.rs

```diff
@@ -1275,6 +1275,31 @@ pub enum InstanceAutoRestartPolicy {
     BestEffort,
 }
 
+// AFFINITY GROUPS
+
+#[derive(Clone, Copy, Debug, Deserialize, Serialize, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum AffinityPolicy {
+    Allow,
+    Fail,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, Serialize, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum AffinityDistance {
```

Comment:
Could rename this to "failure domain", if that seems useful?

Reply:
I like "failure domain", seems more semantically meaningful IMO.

Reply:
Sounds good, updated to "failure domain".